Ally_Desktop_Assistant.py

```diff
@@ -0,0 +1,464 @@
+        start_assistant(query)
+
+
+def written_command_query(event):
+    query = command_var.get()
+    query = query.strip()
+    command_pad.delete(0, END)
+    if not (query == ""):
+        show(f"You : {query}")
+        start_assistant(query)
+    else:
+        pass
+
+
+def give_space(event):
+    text_entered = command_var.get()
+    try:
+        if text_entered[0] != ' ':
+            command_pad.insert(0, ' ')
+    except:
+        pass
+
+
+def exit_it():
+    root.destroy()
+
+
+def add_music_path():
+    music_path = rf"{askdirectory()}"
+
+    try:
+        with open(data_filename) as file_obj:
+            data_dict = json.load(file_obj)
+            data_dict["music_folder_path"] = music_path
+
+        with open(data_filename, 'w') as f_obj:
+            json.dump(data_dict, f_obj)
+
+    except:
+        data_dict = {"voice": "Female", "music_folder_path": music_path}
+        with open(data_filename, 'w') as f_obj:
+            json.dump(data_dict, f_obj)
+
+    tmsg.showinfo('Path Added/Changed', f'Music path added/changed.\n\nCurrent Music Path\n{music_path}')
+
+
+def help_user():
+    tmsg.showinfo("Help", f"This is a desktop assisant named {name}.\nYou can use it to make your work easy.\nTo know more about it, watch the tutorial.")
+
+
+def open_tut():
+    webbrowser.open(tut_address)
+
+
+def open_about():
+    message = f"{name} desktop assistant version 1.0\n\nBuilt on May 17 2020\n\nBuilt by Swastik.\n\nOpen Source Software."
+    tmsg.showinfo('About', message)
+
+
+def show_history():
+    try:
+        history_file = history_filename
+        with open(history_file) as f_obj:
+            history = f_obj.read()
+
+        if history.strip() == "":
+            tmsg.showinfo('History Empty', f'No history found.')
+
+        else:
+            chat_history.delete(0.0, END)
+            chat_history.configure(state=NORMAL)
+            chat_history.insert(END, history)
+            chat_history.yview_pickplace(END)
+            chat_history.update()
+            chat_history.configure(state=DISABLED)
+
+    except:
+        tmsg.showinfo('History Empty', f'No history found.')
+
+
+def clear_history():
+    with open(history_filename, 'w') as file_object:
+        file_object.close()
+
+    chat_history.configure(state=NORMAL)
+    chat_history.delete(0.0, END)
+    chat_history.configure(state=DISABLED)
+
+    tmsg.showinfo('History Cleared', 'History is cleared.')
+
+
+win_color = '#80ff00'
+temp_voice_code = get_voice_code()
+
+root = Tk()
+root.geometry('650x659+10+0')
+root.resizable(0, 0)
+
+root.title(f'{name.title()} Desktop Assistant')
+
+title_label = Label(root, text=f"{name.title()} Desktop Assistant", font="calibri 40 bold",
+                    bg=win_color, fg="blue", padx=2, pady=20)
+title_label.pack(side=TOP, fill=X)
+
+voice_frame = Frame(root, borderwidth=6, bg=win_color)
+voice_frame.pack(fill=X)
+
+Label(voice_frame, text=f"{name}'s Voice", bg=win_color, padx=20,
```

Comment:
Put all the rendering parts inside a render function also make sub-functions and call then if necessary.